cgsmiles/graph_layout_utils.py

```diff
@@ -0,0 +1,368 @@
+from collections import OrderedDict
+import numpy as np
+import scipy.ndimage
+import scipy.optimize
+from scipy.spatial.distance import pdist
+import networkx as nx
+from .linalg_functions import vector_angle_degrees, dihedral_angle_between, rotate_degrees
+
+def _angle(n1, n2, n3, points):
+    ref_vector = points[n3] - points[n2]
+    target_vector = points[n1] - points[n2]
+    angle = vector_angle_degrees(ref_vector, target_vector)
+    return angle
+
+def rotate_subgraph(graph, anchor, reference, target, points, angle=120):
+    """
+    Given a graph of which each node is associated with
+    a point in 2D. Take a `target` edge and remove it
+    from the subgraph. Then take the first connected
+    component and rotate this subgraph by `angle` degrees
+    about the reference point given by the first node in
+    the target edge.
+
+    Parameters
+    ----------
+    graph: nx.Graph
+        the graph
+    target: tuple(abc.hashable, abc.hashable)
+        the target edge
+    points: dict[abc.hashable: np.array]
+        the dict of 2D positions
+    angle: float
+        the angle to rotate by
+
+    Returns
+    -------
+    dict
+        the updated positions dict
+    """
+    graph_copy = nx.subgraph(graph, graph.nodes).copy()
+    graph_copy.remove_edge(anchor, target)
+    connected_comps = nx.connected_components(graph_copy)
+    target_nodes = next(connected_comps)
+    while target_nodes:
+        if target in target_nodes:
+            break
+        target_nodes = next(connected_comps)
+    target_points = np.array([points[node] for node in target_nodes])
+    ref_vector = points[reference] - points[anchor]
+    target_vector = points[target] - points[anchor]
+    rotate_angle = vector_angle_degrees(ref_vector, target_vector) - angle
+    new_points = rotate_degrees(target_points, rotate_angle, origin=points[anchor])
+    for point, node in zip(new_points, target_nodes):
+        points[node] = point
+    return points
+
+def check_and_fix_cis_trans(graph, points):
+    cis_trans = nx.get_node_attributes(graph, 'ez_isomer')
+    for  cis_trans_item in cis_trans.values():
+        for n1, n2, n3, n4, _type in cis_trans_item:
+            angle_init = _angle(n1, n2, n3, points)
+            angle_compl = _angle(n4, n3, n2, points)
+            if _type == 'trans' and np.isclose(angle_init,120, atol=10):
+                continue
+            if _type  == 'cis' and n1 < n4 and np.isclose(angle_init, 120, atol=10):
+                continue
```

Comment:
```suggestion
            if _type == 'trans' and np.isclose(angle_init, 120, atol=10):
                continue
            elif _type  == 'cis' and n1 < n4 and np.isclose(angle_init, 120, atol=10):
                continue
```

Reply:
🤣 pylint says otherwise because the continue prevents you from going to the second if I think

Reply:
True!